scene/animation/animation_manager.h

```diff
@@ -0,0 +1,389 @@
+	friend AnimatedValuesBackup;
+	bool reset_on_save = true;
+	bool editing = false;
+	bool dummy = false;
+#endif // TOOLS_ENABLED
+
+public:
+	enum AnimationCallbackModeProcess {
+		ANIMATION_CALLBACK_MODE_PROCESS_PHYSICS,
+		ANIMATION_CALLBACK_MODE_PROCESS_IDLE,
+		ANIMATION_CALLBACK_MODE_PROCESS_MANUAL,
+	};
+
+	enum AnimationCallbackModeMethod {
+		ANIMATION_CALLBACK_MODE_METHOD_DEFERRED,
+		ANIMATION_CALLBACK_MODE_METHOD_IMMEDIATE,
+	};
+
+	/* ---- Data ---- */
+	struct AnimationLibraryData {
+		StringName name;
+		Ref<AnimationLibrary> library;
+		bool operator<(const AnimationLibraryData &p_data) const { return name.operator String() < p_data.name.operator String(); }
+	};
+
+	struct AnimationData {
+		String name;
+		Ref<Animation> animation;
+		StringName animation_library;
+		uint64_t last_update = 0;
+	};
+
+	struct PlaybackInfo {
+		double time = 0.0;
+		double delta = 0.0;
+		bool seeked = false;
+		bool is_external_seeking = false;
+		Animation::LoopedFlag looped_flag = Animation::LOOPED_FLAG_NONE;
+		real_t weight = 0.0;
+		Vector<real_t> track_weights;
+	};
+
+	struct AnimationInstance {
+		AnimationData animation_data;
+		PlaybackInfo playback_info;
+	};
+
+protected:
+	/* ---- Data lists ---- */
+	LocalVector<AnimationLibraryData> animation_libraries;
+	HashMap<StringName, AnimationData> animation_set; // HashMap<Library name + Animation name, AnimationData>
+
+	TypedArray<StringName> _get_animation_library_list() const;
+	Vector<String> _get_animation_list() const {
+		List<StringName> animations;
+		get_animation_list(&animations);
+		Vector<String> ret;
+		while (animations.size()) {
+			ret.push_back(animations.front()->get());
+			animations.pop_front();
+		}
+		return ret;
+	}
+
+	// For caches.
+	uint64_t animation_set_update_pass = 1;
+	void _animation_set_cache_update();
+
+	// Signals.
+	virtual void _animation_added(const StringName &p_name, const StringName &p_library);
+	virtual void _animation_removed(const StringName &p_name, const StringName &p_library);
+	virtual void _animation_renamed(const StringName &p_name, const StringName &p_to_name, const StringName &p_library);
+	virtual void _animation_changed(const StringName &p_name);
+
+	/* ---- General settings for animation ---- */
+	AnimationCallbackModeProcess callback_mode_process = ANIMATION_CALLBACK_MODE_PROCESS_IDLE;
+	AnimationCallbackModeMethod callback_mode_method = ANIMATION_CALLBACK_MODE_METHOD_DEFERRED;
+	int audio_max_polyphony = 32;
+	NodePath root_node;
+
+	bool processing = false;
+	bool active = true;
+
+	void _set_process(bool p_process, bool p_force = false);
+
+	/* ---- Caches for blending ---- */
+	bool cache_valid = false;
+	uint64_t setup_pass = 1;
+	uint64_t process_pass = 1;
+
+	struct TrackCache {
+		bool root_motion = false;
+		uint64_t setup_pass = 0;
+		Animation::TrackType type = Animation::TrackType::TYPE_ANIMATION;
+		Object *object = nullptr;
+		ObjectID object_id;
+
+		real_t total_weight = 0.0;
+
+		TrackCache() {
+		}
```

Comment:
```suggestion
		TrackCache() { }
```
Though empty constructor and destructor is likely unnecessary.